### PR TITLE
feat(makefile) Use target-specific variable for `RUSTFLAGS`

### PR DESCRIPTION
**Pull request by Hywan**
# Description

Context https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html#Target_002dspecific.

By defining a target-specific variable named `RUSTFLAGS`, the user can
edit it, or set a pre-defined value for it.

Moreover, this patch removes the global `RUSTFLAGS` variable only used
by `link-packages`, and makes it target-specific.

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -708,7 +709,7 @@ install-wasmer-headless-minimal:
 update-testsuite:
 	git subtree pull --prefix tests/wast/spec https://github.com/WebAssembly/testsuite.git master --squash
 
-RUSTFLAGS := "-D dead-code -D nonstandard-style -D unused-imports -D unused-mut -D unused-variables -D unused-unsafe -D unreachable-patterns -D bad-style -D improper-ctypes -D unused-allocation -D unused-comparisons -D while-true -D unconditional-recursion -D bare-trait-objects" # TODO: add `-D missing-docs` # TODO: add `-D function_item_references` (not available on Rust 1.47, try when upgrading)
+link-packages: RUSTFLAGS += "-D dead-code -D nonstandard-style -D unused-imports -D unused-mut -D unused-variables -D unused-unsafe -D unreachable-patterns -D bad-style -D improper-ctypes -D unused-allocation -D unused-comparisons -D while-true -D unconditional-recursion -D bare-trait-objects" # TODO: add `-D missing-docs` # TODO: add `-D function_item_references` (not available on Rust 1.47, try when upgrading)
```
Comment (MarkMcCaskey) [inline Makefile:712]: typo: should be `lint-packages`

**Review by MarkMcCaskey — APPROVED** [review]
Seems fine to me (after we fix the typo)

File: Makefile (hunk 1 above)
Comment (Hywan) [inline Makefile:712]:
```suggestion
lint-packages: RUSTFLAGS += "-D dead-code -D nonstandard-style -D unused-imports -D unused-mut -D unused-variables -D unused-unsafe -D unreachable-patterns -D bad-style -D improper-ctypes -D unused-allocation -D unused-comparisons -D while-true -D unconditional-recursion -D bare-trait-objects" # TODO: add `-D missing-docs` # TODO: add `-D function_item_references` (not available on Rust 1.47, try when upgrading)
```

**Pull request merged**